Android app build failing

I’m having problem building android app. After completing all the steps and then running npm run build:android build fails on
Task :react-native-webrtc:compileDebugJavaWithJavac FAILED

there is a long list of errors all with org.webrtc does not exist

/mattermost-app/node_modules/react-native-webrtc/android/src/main/java/com/oney/WebRTCModule/VideoTrackAdapter.java:5: error: package org.webrtc does not exist

I’m building it on M1 macOS, java 17.0.4.1, gradle 7.6

Build fails with this:

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':react-native-webrtc:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

Beside org.webrtc does not exist there are also this cannot find symbol errors:

symbol:   class SurfaceTextureHelper
  location: class TrackPrivate
/mattermost-app/node_modules/react-native-webrtc/android/src/main/java/com/oney/WebRTCModule/GetUserMediaImpl.java:469: error: cannot find symbol
            MediaStreamTrack track,

There are also some warnings about depricated functions in some of the steps but these above are only errors

--- a/scripts/build.sh
+++ b/scripts/build.sh
@@ -42,6 +42,7 @@ function setup() {
         npm run clean || exit 1
         npm install --ignore-scripts || exit 1
         npx patch-package || exit 1
+        node node_modules/react-native-webrtc/tools/downloadWebRTC.js || exit 1
 
         if [[ "$1" == "ios"* ]]; then
           echo "Installing Gems"

